diff --git a/configure.ac b/configure.ac
index f47f24a189..fb1fdda870 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1747,11 +1747,11 @@ AC_ARG_WITH(linux-input, [  --without-linux-input   don't build linux input even
 
 have_linux_input="no (linux input support disabled)"
 if test "x$with_linux_input" != "xno"; then
-  AC_CHECK_HEADER(linux/input.h,
-	AC_CHECK_DECL(KEY_OK,
-		have_linux_input=yes,
-		have_linux_input="no (needs Linux 2.6)",
-		[#include <linux/input.h>]))
+  AC_CHECK_HEADER([linux/input.h],
+        [AC_CHECK_DECL([KEY_OK],
+                [have_linux_input=yes],
+                [have_linux_input="no (needs Linux 2.6)"],
+                [#include <linux/input.h>])])
 fi
 
 AM_CONDITIONAL(HAVE_LINUX_INPUT, test "x$have_linux_input" = xyes)
